src/components/Reviews/Reviews.js

```diff
@@ -0,0 +1,165 @@
+            carousel.current.scrollWidth - carousel.current.offsetWidth
+          ) {
+            clearInterval(interval);
+            setDirection("backward");
+          }
+          carousel.current.scrollBy({
+            left: carousel.current.offsetWidth / 2,
+            behavior: "smooth",
+          });
+        }
+      }, 2000);
+    }
+  }, [scrollDirection]);
+
+  function handleMouseDown(e) {
+    const x = e.pageX - carousel.current.offsetLeft;
+    setX(x >= 0 ? x : 0);
+    setScrollLeft(carousel.current.scrollLeft);
+    clearInterval(interval);
+  }
+
+  function handleMouseMove(e) {
+    e.preventDefault();
+    if (typeof startX === "number") {
+      carousel.current.style.scrollBehavior = "";
+      carousel.current.style.scrollSnapType = "";
+      setDragging(true);
+      carousel.current.scrollLeft =
+        startX - (e.pageX - carousel.current.offsetLeft) + carouselScrollLeft;
+    }
+  }
+
+  function stopDragging(e) {
+    if (
+      Math.abs(carouselScrollLeft - carousel.current.scrollLeft) < 1 &&
+      e.target !== carousel.current
+    ) {
+      setDragging(false);
+    }
+    setX(false);
+  }
+
+  function handleClick(e) {
+    if (e.target === carousel.current && !isDragging) {
+      carousel.current.style.scrollSnapType = "both mandatory";
+      carousel.current.style.scrollBehavior = "smooth";
+      return;
+    }
+    if (
+      Math.abs(carouselScrollLeft - carousel.current.scrollLeft) < 1 &&
+      e.target !== carousel.current
+    ) {
+      const scrollValue =
+        e.currentTarget.getBoundingClientRect().left -
+        carousel.current.offsetLeft -
+        (carousel.current.offsetWidth - e.currentTarget.getBoundingClientRect().width) / 2;
+
+      carousel.current.scrollBy({
+        left: Math.abs(scrollValue) > 1 ? scrollValue : 0,
+        behavior: "smooth",
+      });
+      setTimeout(() => {
+        carousel.current.style.scrollSnapType = "both mandatory";
+        carousel.current.style.scrollBehavior = "smooth";
+      }, 200);
+    }
+  }
+
+  function handlePress(e) {
+    clearInterval(interval);
+    if (e.type === "keydown" && carousel.current.style.scrollSnapType) {
+      timeout = setTimeout(() => {
+        setDragging(true);
+        carousel.current.style.scrollSnapType = "";
+        carousel.current.style.scrollBehavior = "";
+      }, 300);
+    }
+
+    if (e.type === "keyup" && !isDragging) {
+      clearTimeout(timeout);
+      e.preventDefault();
+      carousel.current.style.scrollSnapType = "both mandatory";
+      carousel.current.style.scrollBehavior = "smooth";
+    }
+  }
+
+  return (
+    <Container>
+      <Text className="title">Reviews</Text>
+      <ReviewsContainer
+        ref={carousel}
+        onMouseDown={handleMouseDown}
+        onMouseMove={handleMouseMove}
+        onMouseLeave={stopDragging}
+        onMouseUp={stopDragging}
+        onClick={handleClick}
+        onKeyUpCapture={handlePress}
```

Comment:
Interesting choice to add left and right key press scrolling. What was your thought process on this?